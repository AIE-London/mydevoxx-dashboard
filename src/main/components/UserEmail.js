import React, { Component } from "react";
import { Redirect } from "react-router-dom";
import retrieveUuid from "../api/retrieveUuid";
import { Form, Button, Input } from "muicss/react";
import { Row } from "react-flexbox-grid";
import styled from "styled-components";

import Card from "./Card";

const LoginPage = styled(Row)`
  height: 100%;
  width: 100%;
  flex: 1;
`;

const LoginCard = styled(Card)`
  margin-top: 2em;
  padding: 2em;
  width: 90%;
  max-width: 400px;
  padding-top: 1em;
  padding-bottom: 0;
  position: relative;
  display: flex;
  flex-direction: row;
  text-align: left;
  padding-bottom: 50px;
  & form {
    width: 100%;
  }
  & button {
    background: transparent;
    right: 0;
    position: absolute;
  }
`;

class UserEmail extends Component {
  constructor(props) {
    super(props);
    this.state = { redirect: false, value: null };
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    alert("An email was submitted: " + this.state.value);

    retrieveUuid
      .getUUID(this.state.value)
      .then(uuid => {
        if (!uuid || uuid.length === 0) {
          throw new Error("Missing UUID");
        }
        this.props.db.record.add({ id: "0", uuid: uuid });
        this.props.onSignIn().then(() => {
          this.setState({ redirect: true });
        });
      })
      .catch(error => {
        this.setState({ error: error });
      });
    event.preventDefault();
    return false;
  }

  render() {
    return (
      <section>
        <LoginPage center="xs" middle="xs">
          <LoginCard>
            <h3> Please login with your email </h3>
            <Form onSubmit={this.handleSubmit}>
              {this.state.redirect && <Redirect to="/" />}
              <Input
                floatingLabel={true}
                required={true}
                type="email"
                name="email"
                label="Email"
                onChange={event => this.setState({ value: event.target.value })}
              />
              {this.state.error &&
                <p>Error: UUID not found, please try again.</p>}
              <Button type="submit">Submit</Button>
            </Form>
          </LoginCard>
        </LoginPage>
      </section>
    );
  }
}

export default UserEmail;
