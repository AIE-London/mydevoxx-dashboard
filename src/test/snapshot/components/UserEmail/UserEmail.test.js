import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import UserEmail from "../../../../main/components/UserEmail";

/*
    Added function to add required functions onto mock DOM node.
    This is necessary to support 'ref' usage in tests.

    By default - Jest will not allow ref usage.
    Issue: https://github.com/facebook/react/issues/7740
 */
function createNodeMock() {
  return {
    focus() {}
  };
}

test("user email component", () => {
  it("will render without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<UserEmail />, div);
  });
});

test("user email component with snapshot", () => {
  const tree = renderer.create(<UserEmail />, { createNodeMock }).toJSON();
  expect(tree).toMatchSnapshot();
});
