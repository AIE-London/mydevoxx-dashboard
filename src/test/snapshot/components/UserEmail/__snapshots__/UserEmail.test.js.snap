exports[`test user email component with snapshot 1`] = `
<section>
  <div
    className="jElMWD row center-xs middle-xs">
    <div
      className="gvZcfI row">
      <h3>
         Please login with your email 
      </h3>
      <form
        className="mui-form "
        onSubmit={[Function]}>
        <div
          className="mui-textfield mui-textfield--float-label "
          style={undefined}>
          <input
            className="mui--is-untouched mui--is-pristine mui--is-empty"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder={null}
            required={true}
            type="email" />
          <label
            onClick={[Function]}
            style={Object {}}>
            Email
          </label>
        </div>
        <button
          className="mui-btn "
          disabled={false}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          type="submit">
          Submit
          <span
            className="mui-btn__ripple-container">
            <span
              className="mui-ripple"
              style={undefined} />
          </span>
        </button>
      </form>
    </div>
  </div>
</section>
`;
